src/psyclone/tests/gocean1p0_stencil_test.py

```diff
@@ -0,0 +1,227 @@
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# ----------------------------------------------------------------------------
+# Author R. W. Ford, STFC Daresbury Lab
+
+'''Stencil tests for PSy-layer code generation that are specific to the
+GOcean 1.0 API.'''
+
+from __future__ import absolute_import
+import os
+import pytest
+from psyclone.parse import parse
+from psyclone.psyGen import PSyFactory
+from psyclone.generator import GenerationError, ParseError
+
+from psyclone.gocean1p0 import GOStencil
+from psyclone import expression as expr
+
+API = "gocean1.0"
+BASE_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                         "test_files", "gocean1p0")
+
+# Section 1
+# Tests for the case where an object of type GOStencil has not been
+# initialised (i.e. the load method has not been called)
+
+
+def test_not_initialised():
+    '''A GOStencil object can be created in isolation and then have its
+    stencil information initialised using the load() method. If a
+    GOStencil object's stencil information has not been initialised
+    then asking for stencil information using the has_stencil, name
+    and depth methods should return an exception. This test checks
+    that an exception is raised as expected.
+
+    '''
+    stencil = GOStencil()
+
+    with pytest.raises(GenerationError) as excinfo:
+        _ = stencil.has_stencil
+    assert "ensure the load() method is called" in str(excinfo.value)
+
+    with pytest.raises(GenerationError) as excinfo:
+        _ = stencil.name
+    assert "ensure the load() method is called" in str(excinfo.value)
+
+    with pytest.raises(GenerationError) as excinfo:
+        _ = stencil.depth(0, 0)
+    assert "ensure the load() method is called" in str(excinfo.value)
+
+# Section 2
+# Tests for the case where the load method in an object of type
+# GOStencil is provided with invalid stencil information
+
+
```

Comment:
I think you're missing a test for the case where the wrong type of object is passed to the `load()` method.

Reply:
I've added tests for passing None and for passing the stencil object itself.

Reply:
Great.